ksqldb-parser/src/main/java/io/confluent/ksql/schema/ksql/SqlValueCoercer.java

```diff
@@ -26,11 +28,71 @@
   /**
    * Coerce the supplied {@code value} to the supplied {@code sqlType}.
    *
-   * <p>Complex SQL types are not supported, (yet).
-   *
    * @param value the value to try to coerce.
    * @param targetSchema the target SQL type.
-   * @return the coerced value if the value could be coerced, {@link Optional#empty()} otherwise.
+   * @return the Result of the coercion.
    */
-  Optional<?> coerce(Object value, SqlType targetSchema);
+  Result coerce(Object value, SqlType targetSchema);
```

Comment:
As above, this change in return value is to allow callers to differentiate between:

* the coercion failed. Previously this returned `Optional.empty` and now returns `Result.failed`.
* the coercion succeeded and the result was `null`. Previously, this threw an NPE and now returns `Result.nullResult`.

Normal results where previously `Optional.of(whateva)` and are now `Result.of(whateva)`.